src/modules/handbook.ts

```diff
@@ -16,53 +16,52 @@ type AlgoliaResult = {
 	url: string;
 };
 
-export class HandbookModule extends Module {
-	constructor(client: CookiecordClient) {
-		super(client);
-	}
+const HANDBOOK_EMBED = new EmbedBuilder()
+	.setColor(TS_BLUE)
+	.setTitle('The TypeScript Handbook')
+	.setURL('https://www.typescriptlang.org/docs/handbook/intro.html')
+	.setFooter({ text: 'You can search with `!handbook <query>`' });
 
-	HANDBOOK_EMBED = new MessageEmbed()
-		.setColor(TS_BLUE)
-		.setTitle('The TypeScript Handbook')
-		.setURL('https://www.typescriptlang.org/docs/handbook/intro.html')
-		.setFooter('You can search with `!handbook <query>`');
-
-	@command({
+export async function handbookModule(bot: Bot) {
+	bot.registerCommand({
+		aliases: ['handbook', 'hb'],
 		description: 'Search the TypeScript Handbook',
-		aliases: ['hb'],
-		single: true,
-	})
-	async handbook(msg: Message, text: string) {
-		if (!text)
-			return await sendWithMessageOwnership(msg, {
-				embeds: [this.HANDBOOK_EMBED],
-			});
-		console.log('Searching algolia for', [text]);
-		const data = await algolia.search<AlgoliaResult>([
-			{
-				indexName: ALGOLIA_INDEX_NAME,
-				query: text,
-				params: {
-					offset: 0,
-					length: 1,
+		async listener(msg, content) {
+			if (!content) {
+				return await sendWithMessageOwnership(msg, {
+					embeds: [HANDBOOK_EMBED],
+				});
+			}
+
+			console.log('Searching algolia for', [content]);
+			const data = await algolia.search<AlgoliaResult>([
+				{
+					indexName: ALGOLIA_INDEX_NAME,
+					query: content,
+					params: {
+						offset: 0,
+						length: 1,
+					},
 				},
-			},
-		]);
-		console.log('Algolia response:', data);
-		const hit = data.results[0].hits[0];
-		if (!hit)
-			return await sendWithMessageOwnership(
-				msg,
-				':x: No results found for that query',
-			);
-		const hierarchyParts = [0, 1, 2, 3, 4, 5, 6]
-			.map(i => hit.hierarchy[`lvl${i}`])
-			.filter(x => x);
-		const embed = new MessageEmbed()
-			.setColor(TS_BLUE)
-			.setTitle(decode(hierarchyParts[hierarchyParts.length - 1]))
-			.setAuthor(decode(hierarchyParts.slice(0, -1).join(' / ')))
-			.setURL(hit.url);
-		await sendWithMessageOwnership(msg, { embeds: [embed] });
-	}
+			]);
+			console.log('Algolia response:', data);
```

Comment:
Stray debug?

Reply:
🤷 it was there before, there are a ton of console logs in places that I'm not sure make sense... the rep module in particular is littered with them.

Reply:
Separate PR I suppose